Flaky end to end test build due to dl.k8s.io not returning

## Description 

The e2e smoke test CI build is [failing repeatedly](https://github.com/kubernetes-sigs/external-dns/actions/runs/21369781166/job/61511034682). The error we see is:


```
Installing kubectl...
curl: Failed to extract a sensible file name from the URL to use for storage
curl: (3) URL using bad/illegal format or missing URL
Error: Process completed with exit code 3.
```

This is due to an http request to get the latest version of kubectl taking forever. From local testing:


```
ime curl -L -s https://dl.k8s.io/release/stable.txt

v1.35.0curl -L -s https://dl.k8s.io/release/stable.txt  0.01s user 0.01s system 0% cpu 1:08.94 total
```

1m to get 7 characters, impressive 😅 

I think there is a throttling mechanism, it smells IP based and the fact that we use managed workers for our github actions jobs isn't helping. 

I am going to hardcode that call for now, see if the downloading part isn't affected in the same or introduce an installation via a package manager that isn't ratelimiting in the same aggressive way.

--- a/scripts/e2e-test.sh
+++ b/scripts/e2e-test.sh
@@ -5,6 +5,7 @@ set -e
 KO_VERSION="0.18.0"
 KIND_VERSION="0.30.0"
 ALPINE_VERSION="3.22"
+KUBECTL_VERSION="1.35.0"
 
 echo "Starting end-to-end tests for external-dns with local provider..."
 
@@ -20,7 +21,7 @@ kind create cluster
 
 # Install kubectl
 echo "Installing kubectl..."
-curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+curl -LO "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl"
 chmod +x kubectl
 sudo mv kubectl /usr/local/bin/kubectl
 

